common/flatpak-utils-http.c

```diff
@@ -447,6 +453,7 @@ flatpak_download_http_uri_once (FlatpakHttpSession    *session,
   g_autofree char *auth_header = NULL;
   g_autofree char *cache_header = NULL;
   g_autoptr(auto_curl_slist) header_list = NULL;
+  g_autoptr(GMutexLocker) curl_lock = g_mutex_locker_new (&session->lock);
```

Comment:
Seems build failed because this is not used.

Reply:
Which is incorrect and the compiler is wrong. But it can just be explicitly used.

Reply:
What do you mean? Where is curl_lock used?

Reply:
It's not meant to be "used". The `GMutexLocker` works in a way that you create it with a mutex, which it locks and keeps it locked until the `GMutexLocker` is freed. As this is an autopointer, the compiler makes sure the variable is freed at the end of the block.

I will add a fake use of the variable at the end as `g_clear_pointer (&curl_lock, g_mutex_locker_free);`, but it's rather unusual and can confuse other people and the static analysers (thinking the variable is not freed on early `return`-s).

Reply:
Based on explanation yes, @TingPing is totally right. Suppressing the warning for this instance sounds legit.

Reply:
This is a gcc/clang behaviour difference. gcc does not warn about this pattern, but the price we pay for that is that gcc also won't warn us if there's an unused `g_autofree char *x = NULL`, `g_autoptr(GError) error = NULL` or similar (in which case the free-function doesn't have any interesting side-effects). clang does warn about this pattern, which is annoying in cases like `GMutexLocker` where the side-effects are the only thing we're interested in, but helpful for other `g_autofree` or `g_autoptr` variables.

Marking it as `G_GNUC_UNUSED` is the way I've usually silenced warnings like this.

For `GMutexLocker` specifically, I wonder whether GLib should automatically put the `G_GNUC_UNUSED` attribute on it (I think that's implementable?), but GLib is now in API freeze, so that isn't going to happen until the 2.75.x cycle even if GLib maintainers think it's desirable.

Reply:
I'd be afraid that `G_GNUC_UNUSED` can have a side-effect to not allocate the variable in some optimization modes, which would mean to not use (in this case) the locking at all, depending on the build options. These things are very hard to debug.

Reply:
Perhaps we should add a new macro to GLib that expands to `__attribute__((__unused__)) __attribute__((__used__))` for this situation? `G_GNUC_USED` or something?

Reply:
That's a question for the glib folks, from my point of view.

The update branch seems to make happy the other compilers.